src/platform/macos/publish.cpp

```diff
@@ -105,7 +105,8 @@ namespace platf::publish {
       &serviceRef,
       0,  // flags
       0,  // interfaceIndex
-      SERVICE_NAME, SERVICE_TYPE,
+      nullptr,  // name
```

Comment:
Since you're giving it a nullptr, it won't use `net::mdns_instance_name()` for macOS at all, and the tests aren't useful for macOS right?

Reply:
Correct, the function is unused on macOS.